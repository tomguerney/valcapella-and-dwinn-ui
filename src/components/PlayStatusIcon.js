import React from 'react'
import CircularProgress from '@material-ui/core/CircularProgress'
import { Pause, PlayArrow } from '@material-ui/icons'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles({
  root: {
    width: 30,
    height: 30
  }
})

const PlayStatusIcon = props => {
  const classes = useStyles()
  if (props.isPlaying && props.isLoading) {
    return <CircularProgress size={20} />
  } else if (props.isPlaying) {
    return <Pause className={classes.root} />
  } else {
    return <PlayArrow className={classes.root} />
  }
}

export default PlayStatusIcon
