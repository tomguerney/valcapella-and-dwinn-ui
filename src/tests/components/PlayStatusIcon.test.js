import React from 'react'
import { shallow } from 'enzyme'
import PlayStatusIcon from '../../components/PlayStatusIcon'

test('should render PlayStatusIcon with play arrow image', () => {
  const wrapper = shallow(<PlayStatusIcon isPlaying={false} isLoading={false} />)
  expect(wrapper.find('PlayArrowIcon').length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})

test('should render PlayStatusIcon with pause image', () => {
  const wrapper = shallow(<PlayStatusIcon isPlaying={true} isLoading={false} />)
  expect(wrapper.find('PauseIcon').length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})

test('should render PlayStatusIcon with circular progress', () => {
  const wrapper = shallow(<PlayStatusIcon isPlaying={true} isLoading={true} />)
  expect(wrapper.find('WithStyles(ForwardRef(CircularProgress))').length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})
