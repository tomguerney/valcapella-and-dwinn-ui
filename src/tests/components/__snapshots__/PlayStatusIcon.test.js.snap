// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayStatusIcon with circular progress 1`] = `
<WithStyles(ForwardRef(CircularProgress))
  size={20}
/>
`;

exports[`should render PlayStatusIcon with pause image 1`] = `
<PauseIcon
  className="makeStyles-root-1"
/>
`;

exports[`should render PlayStatusIcon with play arrow image 1`] = `
<PlayArrowIcon
  className="makeStyles-root-1"
/>
`;
